File: solutions/leetcode_1291.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> list[int]:
        result = []
        low_s = str(low)
        n = len(low_s)
        first = int(low_s[0])
        while True:
            if first + n - 1 > 9:
                first = 1
                n += 1
            num = 0
            for i in range(first, first + n):
                num = num * 10 + i
            if num > high:
                break
            if num >= low:
                result.append(num)
            first += 1

        return result


if __name__ == '__main__':
    given_low = 1000
    given_high = 13000
    expected = [1234, 2345, 3456, 4567, 5678, 6789, 12345]

    solution = Solution()
    result = solution.sequentialDigits(given_low, given_high)
    assert result == expected

File: solutions/leetcode_1291.php
<?php

class Solution {

    /**
     * @param Integer $low
     * @param Integer $high
     * @return Integer[]
     */
    function sequentialDigits($low, $high) {
        $nums = [
            12,23,34,45,56,67,78,89,
            123,234,345,456,567,678,789,
            1234,2345,3456,4567,5678,6789,
            12345,23456,34567,45678,56789,
            123456,234567,345678,456789,
            1234567,2345678,3456789,
            12345678,23456789,
            123456789
        ];
        $result = [];
        foreach ($nums as $num) {
            if ($num < $low) {
                continue;
            }
            if ($num > $high) {
                break;
            }
            $result[] = $num;
        }
        return $result;
    }
}

$givenLow = 1000;
$givenHigh = 13000;
$expected = [1234, 2345, 3456, 4567, 5678, 6789, 12345];

$solution = new Solution();
$result = $solution->sequentialDigits($givenLow, $givenHigh);
assert($result === $expected);
